pkg/ipam/types.go

```diff
@@ -53,6 +56,9 @@ type AllocationResult struct {
 	// InterfaceNumber is a field for generically identifying an interface.
 	// This is only useful in ENI mode.
 	InterfaceNumber string
+
+	// SkipMasquerade indicates whether the datapath should avoid masquerading connections from this IP when the cluster is in tunneling mode.
```

Comment:
We're primarily talking about Pod-to-world connections here, correct? Why does that only affect overlay-routing mode - don't we have to make the same decision ("given a src and dst, should we masquerade?") in native-routing mode?